irida_staramr_results/cli.py

```diff
@@ -0,0 +1,69 @@
+import sys
+import argparse
+import logging
+from version import __version__
+from irida_api import IridaAPI
+from urllib.error import HTTPError
+
+
+def init_argparser():
+    argument_parser = argparse.ArgumentParser(
+        prog="irida-staramr-results",
+        description="Exports StarAMR results available through IRIDA into a single excel report."
+    )
+    argument_parser.add_argument("-v", "--version", action="version", version=f"{argument_parser.prog} {__version__}",
+                                 help="The current version of irida-staramr-results.")
+    argument_parser.add_argument("-p", "--project", action="store",
+                                 help="Project(s) to scan for StarAMR results.")
+    argument_parser.add_argument("-o", "--output", action="store",
+                                 help="The name of the output excel file.")
+
+    # Arguments that will override fields in the config file.
+    argument_parser.add_argument("-u", "--username", action="store",
+                                 help="Override the 'username' in config file. This is your IRIDA account username.")
+    argument_parser.add_argument("-pw", "--password", action="store",
+                                 help="Override the 'password' in config file. This is your IRIDA account password.")
+
+    return argument_parser
+
+
+def main():
+    """
+    Main entry point of irida_staramr_results.
+    Accepts commands form terminal to be processed by the program.
+    """
+    argument_parser = init_argparser()
+
+    if len(sys.argv) == 1:
+        argument_parser.print_help()
+        sys.exit(1)
+
+    args = argument_parser.parse_args()
+    args_dict = vars(args)
+
+    if args_dict["project"] is None:
+        logging.warning("Please specify project(s) to scan for results.")
+    else:
+        irida = IridaAPI("neptune", "6KlqQOEzEy55GBrQdIa28DE9wFk7Y9RkDRmYfCCUKR", "http://10.10.50.155:8080", "admin", "Test123!")
```

Comment:
As a note for later (not in this PR) everything from this line to the end of the try/excepts should be moved to a seperate function in a separate file that contains core logic. This `cli.py` file should only be responsible for argument parsing and user input.

Reply:
Noted. Putting that on the pull request related to configuration.